core/util/src/main/scala/net/liftweb/util/Props.scala

```diff
@@ -48,22 +49,49 @@ import common._
  * The standard Lift properties file extension is "props".
  */
 object Props extends Logger {
+  type PropProvider = scala.collection.Map[String, String]
+
   /**
    * Get the configuration property value for the specified key.
    * @param name key for the property to get
    * @return the value of the property if defined
    */
-  def get(name: String): Box[String] = Box(props.get(name))
+  def get(name: String): Box[String] =
+    sources.map(src => Box(src.get(name)))
+      .find(_.isDefined)
+      .getOrElse(Empty)
+      .map(interpolate)
+
+  private val interpolateRegex = """(.*?)\Q${\E(.*?)\Q}\E([^$]*)""".r
+  private def interpolate(value:Object):String = {
+    def lookup(key:String) = {
+      val tries = interpolators.map(m => Box(m.get(key)))
+      tries.find(_.isDefined).getOrElse(Empty)
+    }
+    val interpolated = for {
+      m <- interpolateRegex findAllMatchIn value.toString
+    } yield {
+      val before   = m group 1
+      val key      = m group 2
+      val after    = m group 3
+      val lookedUp = lookup(key).openOr(value.toString)
+
+      before + lookedUp + after
+    }
```

Comment:
I definitely believe it will, but I'm not sure how to do that with `findAllMatchIn`